### PR TITLE
Feature/unavailable items minicart

### PR DESCRIPTION
**Pull request by lariciamota**
## What's the purpose of this pull request?
Separate items in minicart between available and unavailable as a default behavior. 

## How it works? 
It separates items in minicart in the scenario in which items added to minicart become unavailable after changing the postal code.

## How to test it?
PR in stores

### CODE REVIEW

**Review by FabianaFerreira — APPROVED** [review]

File: packages/gatsby-theme-store/i18n/pt.json [hunk 1]
```diff
@@ -8,6 +8,8 @@
   "minicart.drawer.shipping-disclaimer": "Frete e impostos são calculados no checkout.",
   "minicart.drawer.go-checkout": "FINALIZAR COMPRA",
   "minicart.price.free": "GRÁTIS",
+  "minicart.section.available": "{count, plural, one {# produto disponível} other {# produtos disponíveis}}",
+  "minicart.warning.unavailable": "Este produto não pode ser entregue no endereço fornecido.",
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/i18n/pt.json:12]: Also

File: packages/gatsby-theme-store/src/sdk/product/useAvailability.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react'
+
+import type { OrderForm } from '../orderForm/Provider'
+
+export type OrderFormData = NonNullable<OrderForm>['items']
+
+export interface Product {
+  product: ArrayItem<OrderFormData>
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/sdk/product/useAvailability.ts:8]: where this type ArrayItem come from?

File: packages/gatsby-theme-store/src/sdk/product/useAvailability.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { useMemo } from 'react'
+
+import type { OrderForm } from '../orderForm/Provider'
+
+export type OrderFormData = NonNullable<OrderForm>['items']
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/sdk/product/useAvailability.ts:5]: OrderForm isn't nullable anymore

**Review by emersonlaurentino — APPROVED** [review]

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx [hunk 4]
```diff
@@ -1,65 +1,56 @@
...
-import type { OrderForm } from '../../../sdk/orderForm/Provider'
-import { IMAGE_DEFAULT } from '../../../sdk/product/constants'
+import type { OrderFormData } from '../../../sdk/product/useAvailability'
+import { useAvailability } from '../../../sdk/product/useAvailability'
+import { UnavailableItems } from './ContentItems/UnavailableItems'
+import { AvailableItems } from './ContentItems/AvailableItems'
 
 export interface MinicartContentProps {
-  data: NonNullable<OrderForm>['items']
+  data: OrderFormData
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx:13]: What do you think about inline it `OrderForm['items']` or change the name of this type, maybe or something like this `OrderFormItems`?

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx [hunk 5]
```diff
@@ -1,65 +1,56 @@
...
   variant: string
   imageElement: React.ElementType
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx:15]: Use import type

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/AvailableItems.tsx [hunk 6]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/AvailableItems.tsx:1]: Move to store-ui

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx [hunk 7]
```diff
@@ -1,65 +1,56 @@
 import React from 'react'
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/Content.tsx:1]: move to store-ui

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Move minciart to store ui

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/Item.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+import type { FC } from 'react'
+
+import MinicartDelete from '../Delete'
+import MinicartQuantity from '../Quantity'
+import type { Product } from '../../../../sdk/product/useAvailability'
+
+export interface ItemsProps {
+  items: Product[]
+  variant: string
+  imageElement: React.ElementType
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/Item.tsx:12]: Use named import,
```ts
import type { ElementType } from 'react'
```

File: packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/Item.tsx [hunk 9]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
```
Comment (igorbrasileiro) [inline packages/gatsby-theme-store/src/components/Minicart/Drawer/ContentItems/Item.tsx:1]: move to store ui

**Review by igorbrasileiro — COMMENTED** [review]
Left more comments

File: packages/store-ui/src/Minicart/i18n/pt.ts [hunk 10]
```diff
@@ -5,6 +5,11 @@ export default {
   'minicart.drawer.shipping-disclaimer':
     'Frete e impostos são calculados no checkout.',
   'minicart.drawer.go-checkout': 'FINALIZAR COMPRA',
+  'minicart.price.free': 'GRÁTIS',
```
Comment (klzns) [inline packages/store-ui/src/Minicart/i18n/pt.ts:8]: Make it uppercase with CSS `text-transform: uppercase;`

```suggestion
  'minicart.price.free': 'Grátis',
```

File: packages/store-ui/src/Minicart/Drawer/Content/index.tsx [hunk 11]
```diff
@@ -21,15 +19,19 @@ const useCartSections = <T extends Item>(items: T[]) =>
     () =>
       items.reduce(
         (acc, curr) => {
-          if (curr.sellingPrice === 0) {
-            acc.gifts.push(curr)
+          if (curr.availability === 'available') {
+            if (curr.sellingPrice === 0) {
```
Comment (klzns) [inline packages/store-ui/src/Minicart/Drawer/Content/index.tsx:23]: You can check if an item is a gift reading the property `isGift`

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]

**Review by igorbrasileiro — APPROVED** [review]

**Pull request merged**